src/aiodynamo/utils.py

```diff
@@ -168,9 +168,12 @@ def serialize_dict(value: Mapping[str, Any]) -> Dict[str, Dict[str, Any]]:
 
 
 def parse_amazon_timestamp(timestamp: str) -> datetime.datetime:
-    return datetime.datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%SZ").replace(
-        tzinfo=datetime.timezone.utc
-    )
+    if "." in timestamp:
+        value = datetime.datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
IIRC, Python's datetime parsing thing can parse fractional seconds, but only with three or six fractional digits exactly:
- `59.123` ok
- `59.120` ok
- `59.12` Error

Please collect more test vectors and/or add add more test inputs to cover this corner case.

Reply:
The parser will handle any length from 1 to 6, it just right pads the values. i.e. if AWS were to put `59.12` in the `Expiration` field of the credential response it will be interpreted as `59.120000`

The docs are here:

https://docs.python.org/3/library/datetime.html#format-codes

| Directive | Meaning | Example | Notes |
| --- | --- | --- | --- |
| %f | Microsecond as a decimal number, zero-padded to 6 digits. | 000000, 000001, …, 999999 | (5) |

5. When used with the strptime() method, the %f directive accepts from one to six digits and zero pads on the right. %f is an extension to the set of format characters in the C standard (but implemented separately in datetime objects, and therefore always available).